File: address-geocode-canada-python-examples/get_geo_location_by_municipality_province_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-canada-python/REST"))

from get_geo_location_by_municipality_province_rest import get_geo_location_by_municipality_province


def get_geo_location_by_municipality_province_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n--------------------------------------------------------------------------")
    print("Address Geocode - Canada - GetGeoLocationByMunicipalityProvince - REST SDK")
    print("--------------------------------------------------------------------------")

    municipality = "Ottawa"
    province = "ON"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Municipality: {municipality}")
    print(f"Province    : {province}")
    print(f"License Key : {license_key}")
    print(f"Is Live     : {is_live}")

    try:
        response = get_geo_location_by_municipality_province(municipality, province, license_key, is_live)

        print("\r\n* Geocode Info *\r\n")
        if response and not response.Error:
            print(f"Latitude  : {response.Latitude}")
            print(f"Longitude : {response.Longitude}")
            print(f"PostalCode: {response.PostalCode}")
        else:
            print("No geocode info found.")

        if hasattr(response, "Error") and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Error Message : {str(e)}")

File: address-geocode-canada-nodejs-examples/get_geo_location_by_municipality_province_rest_sdk_example.js
import { GetGeoLocationByMunicipalityProvinceClient } from "../address-geocode-canada-nodejs/REST/get_geo_location_by_municipality_province_rest.js";

async function GetGeoLocationByMunicipalityProvinceRestGo(licenseKey, isLive) {
    console.log("\n--------------------------------------------------------------------------");
    console.log("Address Geocode - Canada - GetGeoLocationByMunicipalityProvince - REST SDK");
    console.log("--------------------------------------------------------------------------");

    const municipality = "Kitchener";
    const province = "ON";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Municipality   : ${municipality}`);
    console.log(`Province       : ${province}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetGeoLocationByMunicipalityProvinceClient.invoke(
            municipality,
            province,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
            return;
        }

        console.log("\n* Geocode Info *\n");
        if (response) {
            console.log(`Latitude  : ${response.Latitude}`);
            console.log(`Longitude : ${response.Longitude}`);
            console.log(`PostalCode: ${response.PostalCode}`);
        } else {
            console.log("No geocode info found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetGeoLocationByMunicipalityProvinceRestGo };
